File: frontend/src/types/socialAdmin.ts
export interface ScenarioRegressionMetrics {
  automation_handled_rate: number;
  llm_fallback_rate: number;
  handoff_rate: number;
  scenario_accuracy: number;
  reference_resolution_accuracy: number;
  product_discovery_accuracy: number;
  unsafe_action_count: number;
  false_order_count: number;
  false_payment_count: number;
}

export interface ScenarioCoverageRow {
  scenario_code: string;
  scenario_name: string;
  description: string;
  supported_providers: string[];
  current_status: string;
  deterministic_handler_exists: boolean;
  LLM_fallback_exists: boolean;
  human_handoff_exists: boolean;
  tests_exist: boolean;
  frontend_support_exists: boolean;
  priority: string;
}

export interface AutomationRuleStep {
  order: number;
  label: string;
  tier: 'deterministic' | 'llm' | 'human';
  detail: string;
}

export interface AdminTask {
  id: string;
  shop_id: string;
  requested_by_user_id: string | null;
  task_type: string;
  input_json: Record<string, unknown>;
  output_json: {
    draft?: string;
    auto_publish?: boolean;
    schema_version?: string;
    task_type?: string;
  };
  status: string;
  requires_approval: boolean;
  approved_by_user_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface OperatorCorrection {
  id: string;
  shop_id: string;
  conversation_id: string;
  message_id: string | null;
  correction_type: string;
  before_json: Record<string, unknown>;
  after_json: Record<string, unknown>;
  operator_id: string | null;
  created_at: string;
}

export interface AutomationSuggestion {
  id: string;
  shop_id: string;
  source_correction_id: string | null;
  suggested_rule_json: {
    type?: string;
    title?: string;
    summary?: string;
    rule?: Record<string, unknown>;
    alias?: Record<string, unknown>;
    test?: Record<string, unknown>;
  };
  status: string;
  created_at: string;
}

export interface OperatorCorrectionInput {
  conversation_id: string;
  message_id?: string;
  before_json: Record<string, string | undefined>;
  after_json: Record<string, string | undefined>;
}

File: backend/app/schemas/social_admin.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class AdminTaskCreate(BaseModel):
    task_type: str = Field(min_length=1, max_length=64)
    context: str = ""
    conversation_id: UUID | None = None


class AdminTaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    shop_id: UUID
    requested_by_user_id: UUID | None
    task_type: str
    input_json: dict[str, Any]
    output_json: dict[str, Any]
    status: str
    requires_approval: bool
    approved_by_user_id: UUID | None
    created_at: datetime
    updated_at: datetime


class OperatorCorrectionCreate(BaseModel):
    conversation_id: UUID
    message_id: UUID | None = None
    before_json: dict[str, Any] = Field(default_factory=dict)
    after_json: dict[str, Any] = Field(default_factory=dict)


class OperatorCorrectionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    shop_id: UUID
    conversation_id: UUID
    message_id: UUID | None
    correction_type: str
    before_json: dict[str, Any]
    after_json: dict[str, Any]
    operator_id: UUID | None
    created_at: datetime


class AutomationSuggestionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    shop_id: UUID
    source_correction_id: UUID | None
    suggested_rule_json: dict[str, Any]
    status: str
    created_at: datetime


class AutomationRuleStepRead(BaseModel):
    order: int
    label: str
    tier: str
    detail: str
